src/config.lua

```diff
@@ -19,10 +20,135 @@
 -- the map this returns into the global config map since that's likely to be the
 -- same everywhere.
 
+-- xxx As of now, answer is the latter. Approaching that problem and deciding
+-- as project further develops. Might do some wonky things until the solution is
+-- decided upon.
+-- xxx Also, keeping track of config[key: changes_abi, value: true/false] is 
+-- going to entail exactly that. Need to sort out how to handle that.
+
 local config = {}
 
 local util = require("util")
 
+-- xxx To be added or deleted as necessary. Keeping here until decision is made.
+config.obsol = {}
+config.unimpl = {}
+config.capenabled = {}
+config.sys_no_abi_change = {}
+config.sys_abi_change = {}
+config.abi_flags = ""
+config.compat_set = {}
+local config_modified = {}
+local cleantmp = true
+-- local tmpspace = "/tmp/sysent." .. unistd.getpid() .. "/"
+
+-- Important boolean keys: configuration file optionally specified and changes 
+-- to the default ABI.
+config.config_file = false
+config.changes_abi = false
+
+-- Default configuration; any of these may get replaced by a configuration file
+-- optionally specified.
+config.default = {
+    abi_func_prefix = "",
+	abi_type_suffix = "",
+    abi_intptr_t = "intptr_t",
+}
+
+-- Each entry should have a value so we can represent ABI flags as a bitmask
+-- for convenience.  One may also optionally provide an expr; this gets applied
+-- to each argument type to indicate whether this argument is subject to ABI
+-- change given the configured flags.
+config.known_abi_flags = {
+	long_size = {
+		value	= 0x00000001,
+		exprs	= {
+			"_Contains[a-z_]*_long_",
+			"^long [a-z0-9_]+$",
+			"long [*]",
+			"size_t [*]",
+			-- semid_t is not included because it is only used
+			-- as an argument or written out individually and
+			-- said writes are handled by the ksem framework.
+			-- Technically a sign-extension issue exists for
+			-- arguments, but because semid_t is actually a file
+			-- descriptor negative 32-bit values are invalid
+			-- regardless of sign-extension.
+		},
+	},
+	time_t_size = {
+		value	= 0x00000002,
+		exprs	= {
+			"_Contains[a-z_]*_timet_",
+		},
+	},
+	pointer_args = {
+		value	= 0x00000004,
+	},
+	pointer_size = {
+		value	= 0x00000008,
+		exprs	= {
+			"_Contains[a-z_]*_ptr_",
+			"[*][*]",
+		},
+	},
+	pair_64bit = {
+		value	= 0x00000010,
+		exprs	= {
+			"^dev_t[ ]*$",
+			"^id_t[ ]*$",
+			"^off_t[ ]*$",
+		},
+	},
+}
+
+config.known_flags = {
+	STD		= 0x00000001,
+	OBSOL		= 0x00000002,
+	RESERVED	= 0x00000004,
+	UNIMPL		= 0x00000008,
+	NODEF		= 0x00000010,
+	NOARGS		= 0x00000020,
+	NOPROTO		= 0x00000040,
+	NOSTD		= 0x00000080,
+	NOTSTATIC	= 0x00000100,
+	CAPENABLED	= 0x00000200,
+	SYSMUX		= 0x00000400,
+
```

Comment:
rather than bits, the known flags might be better as a table so they can be looked up more easily and in a less transliterated from awk way :).